### PR TITLE
refactor(error): clean-up direct error formatting (part 4)

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See backgrounds in #11443. TL;DR: formatting errors with `{}` or `.to_string` can cause the loss of the error source. Use `thiserror_ext::Report` instead.

----

Follow-up of 
- #13763
- #13870
- #14685

This PR fixes usages in storage. The only remaining module is the connector.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/storage/compactor/src/server.rs [hunk 1]
```diff
@@ -414,12 +414,12 @@ pub async fn shared_compactor_serve(
                     tokio::select! {
                         _ = tokio::signal::ctrl_c() => {},
                         _ = &mut shutdown_recv => {
-                                if let Err(err) = shutdown_sender.send(()) {
-                                    tracing::warn!("Failed to send shutdown: {:?}", err);
-                                }
-                                if let Err(err) = join_handle.await {
-                                    tracing::warn!("Failed to join shutdown: {:?}", err);
-                                }
+                            if shutdown_sender.send(()).is_err() {
```
Comment (Li0k) [inline src/storage/compactor/src/server.rs:417]: ditto

**Review by Li0k — APPROVED** [review]
LGTM, thanks for the efforts

File: src/meta/src/hummock/manager/versioning.rs [hunk 2]
```diff
@@ -55,15 +55,12 @@ pub struct HummockVersionSafePoint {
 
 impl Drop for HummockVersionSafePoint {
     fn drop(&mut self) {
-        if let Err(e) = self
+        if self
```
[the remark this answers is not in the document]
Comment (BugenZhao) [reply src/meta/src/hummock/manager/versioning.rs:58]: Yes. The only field in the `SendError` type is the payload itself. As we include the `id` in the log message, there's no extra information.

https://docs.rs/tokio/1.35.1/tokio/sync/mpsc/error/struct.SendError.html

File: src/storage/compactor/src/server.rs (hunk 1 above)
Comment (BugenZhao) [reply src/storage/compactor/src/server.rs:417]: ...if the payload is the unit type, then we can just ignore that.

File: src/meta/src/hummock/manager/versioning.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Li0k) [reply src/meta/src/hummock/manager/versioning.rs:58]: Thanks for reply

**Pull request merged**